### PR TITLE
Corrected the output of latex method in latex.py to return valid latex command

### LINKED ISSUE #15059: not pretty printing indexed(x1, i)
not sure if this is expected behavior but i'm expecting x_{1,i}

image

### PR DESCRIPTION
**Pull request by ayushmankoul**
#### References to other Issues or PRs

Fixes #15059

#### Brief description of what is fixed or changed
The problem was that latex function wasn't printing valid Latex commands for Indexed results.In order to overcome this issue we altered the code to wrap the indexedbase in {} so that printer would print valid latex command.

Example:-

43878942-9caaee84-9b6f-11e8-9f49-44800d684ceb

```
Before Correction in code.
Output:-x_{1}_{i} which was not a valid latex command.

After correction in code.
Output:-{x_{1}}_{i} which is valid latex command.
```
#### Release Notes

* printing
    * fix latex printing of Indexed results #15059

### CODE REVIEW

File: sympy/printing/tests/test_latex.py [hunk 1]
```diff
@@ -522,9 +522,9 @@ def test_latex_indexed():
     Psi_indexed = IndexedBase(Symbol('Psi', complex=True, real=False))
     symbol_latex = latex(Psi_symbol * conjugate(Psi_symbol))
     indexed_latex = latex(Psi_indexed[0] * conjugate(Psi_indexed[0]))
-    # \\overline{\\Psi_{0}} \\Psi_{0}   vs.   \\Psi_{0} \\overline{\\Psi_{0}}
-    assert symbol_latex.split() == indexed_latex.split() \
-        or symbol_latex.split() == indexed_latex.split()[::-1]
+    # \\overline{{\\Psi}_{0}} {\\Psi}_{0}   vs.   \\Psi_{0} \\overline{\\Psi_{0}}
+    assert symbol_latex=='\\Psi_{0} \\overline{\\Psi_{0}}' \
+        or indexed_latex=='\\overline{{\\Psi}_{0}} {\\Psi}_{0}'
```
Comment (jksuom) [inline sympy/printing/tests/test_latex.py]: This should be another assert (no `or`).

File: sympy/printing/latex.py [hunk 2]
```diff
@@ -607,7 +607,9 @@ def _print_BasisDependent(self, expr):
         return outstr
 
     def _print_Indexed(self, expr):
-        tex = self._print(expr.base)+'_{%s}' % ','.join(
+        tex_base = self._print(expr.base)
+        tex_base = '{'+tex_base+'}'
+        tex = tex_base+'_{%s}' % ','.join(
```
Comment (jksuom) [inline sympy/printing/latex.py]: Braces could be added on this line. No need for the one above.

File: sympy/printing/tests/test_latex.py [hunk 3]
```diff
@@ -524,7 +524,7 @@ def test_latex_indexed():
     indexed_latex = latex(Psi_indexed[0] * conjugate(Psi_indexed[0]))
     # \\overline{{\\Psi}_{0}} {\\Psi}_{0}   vs.   \\Psi_{0} \\overline{\\Psi_{0}}
     assert symbol_latex=='\\Psi_{0} \\overline{\\Psi_{0}}' \
-        or indexed_latex=='\\overline{{\\Psi}_{0}} {\\Psi}_{0}'
+        and indexed_latex=='\\overline{{\\Psi}_{0}} {\\Psi}_{0}'
```
Comment (jksuom) [inline sympy/printing/tests/test_latex.py]: This should not be a continuation line with `and` but `assert indexed_latex == ...`. The results are now different and each one should have its own test line.

File: sympy/printing/tests/test_latex.py [hunk 4]
```diff
@@ -522,9 +522,9 @@ def test_latex_indexed():
     Psi_indexed = IndexedBase(Symbol('Psi', complex=True, real=False))
     symbol_latex = latex(Psi_symbol * conjugate(Psi_symbol))
     indexed_latex = latex(Psi_indexed[0] * conjugate(Psi_indexed[0]))
-    # \\overline{\\Psi_{0}} \\Psi_{0}   vs.   \\Psi_{0} \\overline{\\Psi_{0}}
-    assert symbol_latex.split() == indexed_latex.split() \
-        or symbol_latex.split() == indexed_latex.split()[::-1]
+    # \\overline{{\\Psi}_{0}} {\\Psi}_{0}   vs.   \\Psi_{0} \\overline{\\Psi_{0}}
+    assert symbol_latex=='\\Psi_{0} \\overline{\\Psi_{0}}'
+    assert indexed_latex=='\\overline{{\\Psi}_{0}} {\\Psi}_{0}'
```
Comment (sidhantnagpal) [inline sympy/printing/tests/test_latex.py]: SymPy uses spaces around `==`.

Also, it would be good to have meaningful commit messages.

**Comment by ayushmankoul** [comment]
Thank You @jksuom for pointing out my mistakes and helping me to fix it.I have made the necessary changes as advised by you and other contributors.Is the description of pull request fine now or should I also add the expected output of @majidaldo inside the description ? I am new to open source contributions and I hope am able to contribute to organisation as much I can.

**Comment by jksuom** [comment]
Please add a test for `latex(Indexed('x1', Symbol('i')))`.

File: sympy/printing/tests/test_latex.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from sympy import (
     Add, Abs, Chi, Ci, CosineTransform, Dict, Ei, Eq, FallingFactorial,
-    FiniteSet, Float, FourierTransform, Function, IndexedBase, Integral,
+    FiniteSet, Float, FourierTransform, Function,Indexed, IndexedBase, Integral,
```
Comment (jksuom) [inline sympy/printing/tests/test_latex.py]: Please add a space after comma.

Comment (ayushmankoul) [reply sympy/printing/tests/test_latex.py]: @jksuom Sure sir I have updated the code as asked by you.

**Comment by jksuom** [comment]
This looks good. Thank you for your contribution.

**Pull request merged**